### ISSUE TITLE
Failed example 3 wrong for 'role attribute has valid value'

### ISSUE THREAD

**Issue by dd8** [act-rules/act-rules.github.io#1131]
## Failed example 3
```
<input type="text" role=" " />
```
https://act-rules.github.io/rules/674b10#failed-example-3

## Inapplicable example 1
```
<div role="">Some Content</div>
```

The roles in these examples are both equivalent since the value is space separated tokens, so both should be inapplicable or both failed. See step 1 B here:
https://www.w3.org/TR/wai-aria-implementation/#mapping_role

Both of these produce the same empty list of roles.

ARIA 1.2 does define token list as:
> A list of one or more tokens
https://www.w3.org/TR/wai-aria-1.2/#host_general_role

This is unlikely to be harmful since ARIA also says only the first valid, concrete role should be applied:
>If the role attribute contains no tokens matching the name of a non-abstract WAI-ARIA role, the user agent MUST treat the element as if no role had been provided
https://www.w3.org/TR/wai-aria-1.2/#document-handling_author-errors

...but it is classed as an ARIA authoring error...

*Edit:*  ... which is an ARIA conformance issue, but not guaranteed to be a WCAG conformance issue (especially if devs have randomly sprinkled role='' over their code). Generally valid roles used inappropriately are a much worse problem. For example:
```
<body role='presentation'>
````

*Note:* ARIA doesn't define which characters are whitespace, but the actual algorithm used by browsers is very likely to be this one since browsers already use this for `class` and `autocomplete` attributes:
https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#space-separated-tokens

**Update:** HTML is very inconsistent regarding empty vs whitespace-only attributes vs leading and trailing whitespace:

`alt=''` is not the same as `alt='  '` 
`lang=''` is not the same as `lang='  '` 
`id='test'` is not the same as `lang=' test '` 
`class=''` is the same as `class='  '` 
`class='test'` is the same as `class=' test '` 
`role=''` is the same as `role=' '` 
`role='img'` is the same as `role=' img '`

**Comment by WilcoFiers** [comment]
Yeah, that inconsistency in HTML isn't great. I'm not sure there's actually an issue here. This rule doesn't test conformance to ARIA but to WCAG, and for WCAG it doesn't matter if the role attribute is valid or not, or at least it doesn't for 4.1.2 (I'm sure some people will argue 4.1.1, but I wouldn't).

Either way, the reason i think this should be what it is is that things like that can happen rather easily through templating languages, for example:

```
<div role="`${role || 'button'}`">
```

If you want to force code like that not to express a button role, just do `role = ' '`. Similarly without the `button` default, code like that could end up with an empty role attribute. Stuff like that can be done thoughtfully, without causing problems for AT.

**Comment by dd8** [comment]
Just did some testing - specifying `role=""` or `role="   "` or omitting the `role` attribute produces identical results in the accessibility tree - that means Failed Example 3 isn't a WCAG failure:

https://act-rules.github.io/rules/674b10#failed-example-3

It is an ARIA conformance failure but it isn't a WCAG 4.1.2 failure.

These all produce an identical textbox accessible object in Chrome/Safari/Firefox a11y tree:

```
    <input type="text" role=" " />
    <input type="text" role="" />
    <input type="text" />
```

These all produce an identical text node accessible object in Chrome/Safari/Firefox a11y tre:

```
    <div role=" " >text</div>
    <div role="" >text</div>
    <div >text</div>
```

These all produce an identical button accessible object in Chrome/Safari/Firefox a11y tree:

```
    <button role=" " ></button>
    <button role="" ></button>
    <button ></button>    

```

**Comment by WilcoFiers** [comment]
Yeah, I would expect as much. OK this needs to be fixed. `role=" "` shouldn't fail that rule IMO.

**Comment by Jym77** [comment]
Looks like updating applicability to "any role attribute that is not only ASCII whitespace" would do the trick. (that would mean the list of space separated tokens contains at least one token and it make sense to check whether one of these is a valid role).

**Comment by Jym77** [comment]
Closed by #1277.
